src/AbstractFixer.php

```diff
@@ -34,7 +43,7 @@ public function __construct()
     {
         if ($this instanceof ConfigurableFixerInterface) {
             try {
-                $this->configure(null);
+                $this->configure(array());
```

Comment:
I pointed it here to show it.
`getConfigurationDefinition` must not create new instance all the time it is called. (Especially with new approach of creating bunch of DTO objects).

Currently, we generate new instances for one fixer when:
1. class is created, default configuration is applied
2. class is configured with user-provided config

No need in real tool run for doubling the object instantation.
Then, we also test it:
3. With every tests execution in `setUp`
4. With some tests execution when we provide custom config.

Also, applying the default config should be ultra fast. 

---------------

Proposal: lazy loading.
in whole src/fixers/*/ rename `getConfigurationDefinition` to `createConfigurationDefinition` and add new `getConfigurationDefinition` to this abstract class that will use `createConfigurationDefinition` if the instance was not yet created.